### PR TITLE
Add basic LSP client

### PR DESCRIPTION
**Pull request by antroseco**
I think this is sufficient to get us started. The actual implementation is delegated to the official implementation; we just need to configure it.

### CODE REVIEW

File: src/lsp_client.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        "Graphene",
+        server_options,
+        client_options
+    );
+
+    // Start the client. This will also launch the server.
+    return client.start();
+}
+
+function stop() {
```
Comment (DaveDuck321) [inline src/lsp_client.js]: So `client.stop()` returns a promise right? Any reason not to mark it async?

File: src/lsp_client.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        // Bad path, this check is for the user's convenience only.
+        vscode.window.showErrorMessage(`Invalid language server path: '${path}'`);
+        return;
+    }
+
+    /** @type {language_client.ServerOptions} */
+    const server_options = {
+        run: {
+            command: path,
+            transport: language_client.TransportKind.pipe,
```
Comment (DaveDuck321) [inline src/lsp_client.js]: How can we log debug errors in pipe transport? Would std error get correctly reported by vscode?

File: src/lsp_client.js (hunk 1 above)
Comment (DaveDuck321) [reply src/lsp_client.js]: Actually shouldn't every function in this file be marked async?

**Review by DaveDuck321 — APPROVED** [review]
Can't wait for the implementation 😎😎😎.

File: src/lsp_client.js (hunk 1 above)
Comment (antroseco) [reply src/lsp_client.js]: It's not required, but I can mark them anyway

File: src/lsp_client.js (hunk 2 above)
Comment (antroseco) [reply src/lsp_client.js]: Yep, if you look at the code I just print to stdout and that shows up in the `Output` tab of vscode, under `Graphene`.

**Pull request merged**